### PR TITLE
Share Extension: Improve text import from other apps

### LINKED ISSUE #11222: [Sharing] Share Extension: Improve text import from other apps
When sharing text to our share extension from other apps (I tested Bear and Ulysses), one of two things happens:

1. The share extension is launched with the shared text inserted as a block quote
2. The share extension is launched with an empty editor

In Bear, to reproduce number 1, simply tap the system share icon in the top right of a note. To reproduce number 2, tap the (i) icon in the top right of a note, tap TXT at the bottom, and attempt to share the text version of the note.

There are two issues here:

### Text inserted as a blockquote

This is a conscious choice we made when we first built the share extension. If only text is being shared to the extension, we insert it as a blockquote. One use case for this is for example sharing snippets of text from a web page (highlight the text in Safari, and choose Share from the popup menu). It seems that we have no way to detect whether a user is sharing selected text or an entire document. 

I think we have a couple of options here, and I'd be interested to get input from @wordpress-mobile/automattic-design:

* Revisit this decision, and perhaps never auto-quote shared text
* Auto-quote text only if it's below a certain character / word count (but how do you decide the limit?)
* Allow the user to choose whether imported text is quoted or not. How would we surface this? A setting in the main app's App Settings?

### No text inserted

I saw the same behaviour from both Ulysses and Bear, and it looks like it's a bug on our end. In this situation, the `NSItemProviders` sent to the share extension contain two types of content: `public.file-url` (a URL for a local file containing the shared content), and (in Bear's case, but not Ulysses') the shared content as `public.plain-text`.

When we attempt to extract shared content (in ShareExtractor.swift), we [iterate through a number of different extractors that we've written](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/WordPressShareExtension/ShareExtractor.swift#L126). When we reach an extractor that can extract valid content, we stop checking the other extractors.

In this case, the URL extractor says it can extract the URL content, so we never look at the text. However, the URL extractor [bails out](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/WordPressShareExtension/ShareExtractor.swift#L233) later if it detects that the URL is a file URL. So we end up with no content.

There are a couple of solutions here:

* Make the parsing / extracting more robust, and perhaps prefer text content over URL content if it's available.
* Add the ability for the URL extractor to copy content from a file URL

On top of each of these fixes, we could also look at supporting other content types in the share extension, such as JSON or TextBundle – but we can open separate issues for that.

cc @nheagy @matteorattotti

### LINKED ISSUE #11270: Share Extension: Can be crashed with weird HTML
I believe this may be the cause of #10263

### Steps to reproduce the crash:

To crash the app:
- Go to Notes.app
- Type in ```this will <crash>crash</crash> the share extension```
- Share to WordPress using the share extension
- Nothing appears (the extension crashes).

### Why this happens

Currently, plain text sent to the share extension is treated as HTML, and wrapped in `<blockquote>`. You can see this by sharing "```my <em>note</em>```" from Notes.app. It will appear in our share extension as:

> my _note_

So when sharing the crashing string above, the Aztec instance within the share extension doesn't know what to do with the `<crash>` tag, and crashes.

There are two components to this issue:

1. Incoming plain text is treated as HTML, but instead its HTML entities should be escaped
2. Aztec within the share extension is unprepared for weird HTML

The latter of which will be important if the share extension ever intends to support HTML import.

### PR DESCRIPTION
**Pull request by nheagy**
Fixes #11222
Fixes #11270 

This fixes text import from other applications by treating text from shared files differently than from text shared directly (which will still be blockquoted). It also fixes the issue of shared text (both kinds) being treated like HTML, opening the possibility of nefarious HTML making it into the user's site.

text import longer

To test:
- Share text from bear using the export UI. See GIF. (cc: @trix180)
- Share text from Ulysses using text UI
- In both cases, ensure that the text appears in the body of the post without block quote
- Also trying sharing from other text editors you may have. Some may only share the text directly, not a file, and the blockquote will still appear
- Also, try sharing HTML source in another editor. Now that HTML should appear with escaped HTML entities.

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/WordPressShareExtension/ShareExtractor.swift [hunk 1]
```diff
@@ -195,6 +205,7 @@ private extension TypeBasedExtensionContentExtractor {
 
     func extract(context: NSExtensionContext, completion: @escaping ([ExtractedItem]) -> Void) {
         let itemProviders = context.itemProviders(ofType: acceptedType)
+        print(acceptedType)
```
[the remark this answers is not in the document]
Comment (nheagy) [reply WordPress/WordPressShareExtension/ShareExtractor.swift]: Ah, I meant to remove that entirely 😉

**Comment by nheagy** [comment]
@frosty ready for re-review!

**Comment by trix180** [comment]
This is hands down awesome.
We are working with some of the team to a custom sharing of a TextBundle format with the Wordpress app which will allow also in-place attachments and images.

About the HTML, I may be missing a use case here but I wonder if escaping it the correct approach instead of converting it to an attributed string and using it in the editor. This approach will eventually allow passing images as base64 in the HTML document.

**Comment by frosty** [comment]
@trix180 Thanks for the HTML –> Attributed String idea... we'll investigate!

**Comment by frosty** [comment]
Merging as Nate is AFK this week.

**Pull request merged**